spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/telemetry/TelemetryTracker.java

```diff
@@ -42,11 +44,27 @@ public TelemetryTracker(Azure azure, String resourceGroup) {
         this.defaultProperties.put(PROPERTY_INSTALLATION_ID, TelemetryUtils.getHashMac());
     }
 
+    private TelemetryClient getTelemetryClient(TelemetryProperties telemetryProperties) {
+        final String instrumentationKey = telemetryProperties.getInstrumentationKey();
+
+        if (StringUtils.hasText(instrumentationKey)) {
```

Comment:
If instrument key has special format, do we need to validate here? If passed wrong key, will application insight sdk report error? Also, we should log validation error here, otherwise we're unaware of such error.

Reply:
I am not sure if there is a format here. insight sdk take care the invalid key silently. As the key is only available around use, I think we may not need to handle that at all.